sqlx-cli/src/bin/cargo-sqlx.rs

```diff
@@ -9,6 +10,7 @@ async fn main() {
     // so we want to notch out that superfluous "sqlx"
     let args = env::args_os().skip(2);
 
+    dotenv().ok();
```

Comment:
```suggestion
    let _ = dotenv();
```

Using `.ok()` to discard errors is not particularily idiomatic. This [would have been fixed](https://github.com/dotenv-rs/dotenv/pull/57) in dotenv's examples already if it was actively maintained.

Reply:
@jplatte thanks for your suggestion. Some follow up questions:

To what degree is this a style concern? A behavior concern?

> Using `.ok()` to discard errors is not particularly idiomatic.

Why?

In particular, has the Clippy community discussed this? Has there been a consensus or something close to it? Here's what I've found that is somewhat related: [Warn about blanket let _ =](https://github.com/rust-lang/rust-clippy/issues/6842)

I don't have a very strong preference here, but I lean towards preferring using `ok()` as above. For example, I think I agree with this part of [tshepang's comment](https://github.com/dotenv-rs/dotenv/pull/57#issuecomment-886177461):

> I ... think `dotenv.ok()` makes it clear enough that failure is acceptable

P.S. If you are interested in how I think about this more generally, see [What does 'idiomatic' mean in Rust?](https://gist.github.com/xpe/c8eedf8123285eb763b80d788eeac8c2)

Reply:
Behavior is the exact same, this is just a style concern. I've left a comment on the clippy issue you posted with my arguments for `let _ = <expr>;` over `<expr>.ok();`.

Reply:
Found another issue: https://github.com/rust-lang/rust/issues/66116

Reply:
We use `.ok()` a lot internally; IMO it's quicker to use because it's auto-suggested by my IDE as soon as I type `.o` which is personally important to me when setting up boilerplate like this.

In the discussions linked here, it seems like the jury is still out on the issue of what's idiomatic and what's not in terms of discarding results, so I don't think it's necessary to spend much time bikeshedding. Just use `.ok()` for now and if that method gains a `#[must_use]` later, or Clippy starts linting against it, then we can `cargo fix` that easily.